### DISCUSSION TITLE
Can this be used outside of components too?

### DISCUSSION THREAD

**Discussion by dietergeerts** [TanStack/query#5200] [Q&A]
I'm validating if we can use this library for our store in a Vue application. But what I see is that this library is mainly used through hooks, which is fine when you want data in your components at route level. Of course, there are lots of use-cases where you have code that needs to use the store that doesn't involve rendering components, like doing some socket handling. So is it possible to execute queries to the store and use them outside of components?

**Comment by TkDodo** [comment]
sure - with `queryClient.fetchQuery`
